src/Chainweb/Pact/PactService/Pact5/ExecBlock.hs

```diff
@@ -493,11 +494,21 @@ validateParsedChainwebTx _logger v cid db _blockHandle txValidationTime bh isGen
   | otherwise = do
       checkUnique tx
       checkTxHash tx
+      checkChain
       checkTxSigs tx
       checkTimes tx
       return ()
   where
 
+    checkChain :: ExceptT InsertError IO ()
+    checkChain
+      | Pact5.assertChainId cid txCid =
+        return ()
+      | otherwise =
+        throwError $ InsertErrorWrongChain (chainIdToText cid) (Pact5._chainId txCid)
+      where
+      txCid = view (Pact5.cmdPayload . Pact5.payloadObj . Pact5.pMeta . Pact5.pmChainId) tx
```

Comment:
Seems like this could be combined with the `Pact4` check so we're not repeating ourselves. Maybe just pass the lens in as a param and throw it in `Validations` or `Utils`? The Pact4 version doesn't need the viewpatterns per se.